cmake/ext_project_autotools.cmake

```diff
@@ -0,0 +1,118 @@
+include(utils)
+
+# Build a third-party project that uses autotools to generate build system.
+# Basically it generates configure script, run configure and then make and
+# rerun them according to their dependants in case of incremental build.
+#
+# DIR - relative path to project directory in main project tree.
+# CONFIGURE - options for configure script.
+# BYPRODUCTS - products caller targets depend upon.
+#
+# Note that incremental rebuild in case autotools files or config files are
+# changed is available only if CMake version is at least 3.12. Otherwise
+# rebuild will not be triggered. Rebuild is only required for developers
+# which presumably has newer CMake versions.
+function(ext_project_autotools name)
+    set(sv_keywords
+        DIR
+    )
+    set(mv_keywords
+        CONFIGURE
+        BYPRODUCTS
+    )
+    set(options
+        ""
+    )
+    cmake_parse_arguments(
+        ARGS "${options}" "${sv_keywords}" "${mv_keywords}" ${ARGN}
+    )
+    if (NOT DEFINED ARGS_DIR)
+        message(FATAL_ERROR "DIR argument is mandatory")
+    endif()
+
+    # CONFIGURE_DEPENDS available since 3.12
+    # VERSION_GREATER_EQUAL is only since 3.7
+    if (${CMAKE_VERSION} VERSION_GREATER "3.12" OR
+        ${CMAKE_VERSION} VERSION_EQUAL "3.12")
+        set(autotool_files_standard
+            configure.ac
+            acinclude.m4
+        )
+        list_add_prefix(
+            autotool_files_standard "${ARGS_DIR}/" autotool_files_standard
+        )
+
+        file(
+            GLOB_RECURSE autotool_files_am
+            CONFIGURE_DEPENDS
+            RELATIVE "${CMAKE_SOURCE_DIR}"
+            "${ARGS_DIR}/*.am"
+        )
+
+        set(autotool_files ${autotool_files_am} ${autotool_files_standard})
+
+        file(
+            GLOB_RECURSE config_files_in
+            CONFIGURE_DEPENDS
+            RELATIVE "${CMAKE_SOURCE_DIR}"
+            "${ARGS_DIR}/*.in"
+        )
+
+        # Makefile is used as dependency between configure and make thus we
+        # need to exclude it from dependency for config.status step as
+        # it cannot be output of two custom commands.
+        list(REMOVE_ITEM config_files_in ${ARGS_DIR}/Makefile.in)
```

Comment:
The command is about `Makefile.in`, but the comment is about `Makefile`. TBH, I don't get the idea.

Reply:
This is a remnant from draft version where output of `config.status` custom command was list of `${config_files_in}` counterparts. So that root `Makefile` was output of both `config.status` and `configure` custom commands without this hunk which is invalid configuration.

Removed this hunk.